controllers/admin/AdminImportController.php

```diff
@@ -1711,23 +2315,25 @@ protected function productImportOne($info, $default_language_id, $id_lang, $forc
                 $supplier->name = $product->supplier;
                 $supplier->active = true;
 
-                if (($field_error = $supplier->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-                    ($lang_field_error = $supplier->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
-                    !$validateOnly &&  // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
-                    $supplier->add()) {
+                // FIXME needing such a comment about "&& !$validateOnly" position is a code smell. Refacto needed.
+                if (($fieldError = $supplier->validateFields(UNFRIENDLY_ERROR, true)) === true
+                    && ($langFieldError = $supplier->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true
+                    && !$validateOnly  // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
+                    && $supplier->add()
+                ) {
                     $product->id_supplier = (int)$supplier->id;
                     $supplier->associateTo($product->id_shop_list);
                 } else {
                     if (!$validateOnly) {
-                        $this->errors[] = sprintf(
-                            $this->trans('%1$s (ID: %2$s) cannot be saved', array(), 'Admin.Advparameters.Notification'),
+                        $this->errors[] = $this->trans('%1$s (ID: %2$s) cannot be saved', array(
```

Comment:
Same comment about translation placeholder. Please use a variable name surrounded by `%%` an add it in the second parameter.